pandas/core/arrays/datetimes.py

```diff
@@ -1747,12 +1748,78 @@ def _generate_range_overflow_safe(endpoint, periods, stride, side='start'):
     """
     # GH#14187 raise instead of incorrectly wrapping around
     assert side in ['start', 'end']
+
+    i64max = np.iinfo(np.int64).max
+    msg = ('Cannot generate range with {side}={endpoint} and '
```

Comment:
Mentioned in https://github.com/pandas-dev/pandas/issues/24252#issuecomment-446729164, would it be a good opportunity to change this error message so we display is date being overflowed instead of the nanosecond epoch time (since that was mostly likely the original argument)?

Reply:
Good idea